mantidimaging/core/data/test/image_stack_test.py

```diff
@@ -240,3 +240,50 @@ def test_name_unique_exists(self):
         imgs = ImageStack(np.asarray([1]), name="tomo")
         imgs.make_name_unique(existing)
         self.assertEqual(imgs.name, "tomo_2")
+
+    def test_slice_as_stack(self):
+        raw_pixels = np.arange(60, dtype=np.float32).reshape((3, 4, 5))
+        image = ImageStack(raw_pixels.copy(), name="tomo", sinograms=False)
+
+        np.testing.assert_array_equal(raw_pixels[[0], :, :], image.slice_as_image_stack(0).data)
+        np.testing.assert_array_equal(raw_pixels[[2], :, :], image.slice_as_image_stack(2).data)
+        self.assertRaises(IndexError, image.slice_as_image_stack, 3)
+
+        slice = image.slice_as_image_stack(0)
+        self.assertEqual(slice.height, image.height)
+        self.assertEqual(slice.width, image.width)
+        self.assertEqual(slice.num_projections, 1)
+
+        # check that modification do not affect original
+        copy = image.slice_as_image_stack(0)
+        copy.data += 1
+        np.testing.assert_array_equal(raw_pixels, image.data)
+
+        image = ImageStack(raw_pixels.copy(), name="tomo", sinograms=True)
+        slice = image.slice_as_image_stack(0)
+        self.assertEqual(slice.height, 1)
+        self.assertEqual(slice.width, image.width)
+        self.assertEqual(slice.num_projections, image.num_projections)
+        np.testing.assert_array_equal(raw_pixels[[0], :, :], slice.data)
+
+    def test_sino_as_stack(self):
+        raw_pixels = np.arange(60, dtype=np.float32).reshape((3, 4, 5))
+        image = ImageStack(raw_pixels.copy(), name="tomo", sinograms=False)
+
+        print(image.sino_as_image_stack(0).data.shape)
```

Comment:
Was this print statement mean to stay in?

Reply:
No, i'll remove that.